### ISSUE TITLE
SDK is returning Authentication failed When User is in Forced password/ Password reset state.

### ISSUE THREAD

**Issue by Sushree-nayak-2022** [okta/okta-idx-swift#123]
### Describe the bug?

When User is in Forced password/ Password reset state. Sometimes SDK is detecting correctly and returning password reset state but after doing multiple rounds of testing found out many times SDK is returning Authentication failed state.

Steps to reproduce on OKTA-IDX Sample app:

- Login into the app.
- Call /api/v1/users/UserID/lifecycle/expire_password?tempPassword=true to set a temp password for the user via Postman.
- Login into the app again.
- Sometimes We will get Authentication failed and sometimes we will get correct state which is password reset state.

### What is expected to happen?

SDK should always return Forced password state if user is in password reset state.
I check that by using remediations[.reenrollAuthenticator].

### What is the actual behavior?

SDK should return Forced password state which is happening few times. I check that by using remediations[.reenrollAuthenticator]. 
But Many times SDK just returns authentication failed and remediations does not contain reenrollAuthenticator.
As a result we are not asking user to reset their password instead we are showing authentication failed which is misleading because User may think the credential is wrong.

### Reproduction Steps?

Steps to reproduce on OKTA-IDX Sample app:

Login into the app.
Call /api/v1/users/UserID/lifecycle/expire_password?tempPassword=true to set a temp password for the user via Postman.
Login into the app again.
Sometimes We will get Authentication failed and sometimes we will get correct state which is password reset state.

### SDK Version(s)

3.0.5

**Comment by mikenachbaur-okta** [comment]
I believe the response you get may depend upon the workflow being followed. For the most part, expired or reset passwords should enable the user to step through the authentication process and reenrol their password within the SDK.

If you deactivate and then reactivate the user, they will receive an "Authentication Failed" error until they activate their account using the activation email. 

There are some circumstances where you will receive an error indicating the "Flow continued on another tab".  This may occur when authentication begins in the mobile application, but an activation email takes them to enrol their password within the web browser.  When this occurs, the authentication session on the mobile client will terminate.  If you're encountering this situation, you would need to restart the sign in process on the mobile client.

Did this answer your questions @Sushree-nayak-2022? I'd be more than happy to go into this in more detail, especially if I knew more about your specific policy settings which may be influencing the sign in flow.

**Comment by mikenachbaur-okta** [comment]
@Sushree-nayak-2022 One thing I did notice is that the API will often return HTTP status codes >= 400 accompanied by a proper JSON payload. However, this status code was resulting in the SDK throwing an error response, instead of allowing the client to render UI to address the error.

I've just released [OktaIdx 3.0.6](https://github.com/okta/okta-idx-swift/releases/tag/3.0.6) which ensures valid IDX responses can still be processed by clients, even when the HTTP status code indicates an error occurred.

I'm not sure if this will unblock you, but I'd suggest you give it a try to see if it helps.

**Comment by Sushree-nayak-2022** [comment]
Thanks @mikenachbaur-okta , Thanks for the update.
I can still see the issue.

This is what I did on IDX sample app(This is just an example. sometime authentication failed comes on 2nd try as well.)

- User state now is PASSWORD EXPIRED
- Login with username and temp password -> Detected its pass expired case. But I  will not change the password at this point.I will cancel 
- Login again with username and temp password -> Detected its pass expired case. But I  will not change the password at this point.I will cancel 
- Login again with username and temp password -> Detected its pass expired case. But I  will not change the password at this point.I will cancel 
- Login again with username and temp password -> Authentication failed

I checked if account is locked at this point but the account status was still showing PASSWORD EXPIRED
